extensions/api/auth-basic/src/main/java/org/eclipse/dataspaceconnector/api/auth/BasicAuthenticationExtension.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright (c) 2022 ZF Friedrichshafen AG
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       ZF Friedrichshafen AG - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.auth;
+
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.system.Provides;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+import static java.lang.String.format;
+
+/**
+ * Extension that registers an AuthenticationService that uses API Keys
+ */
+@Provides(AuthenticationService.class)
+public class BasicAuthenticationExtension implements ServiceExtension {
+
+    @EdcSetting
+    public static final String BASIC_AUTH = "edc.api.auth.basic";
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var basicAuthUsers = context.getConfig(BASIC_AUTH).getRelativeEntries();
```

Comment:
I've give it a try. But now I got this problem `stream has already been operated upon or closed` because the stream will be processed every time a request comes in which isn't possible. To fix it the internet recommends to use a `Supplier<Stream<Config>>` which in my eyes is more complex than just using the map approach.

Reply:
Yes, you'll need to collect the stream before passing it to the service

Reply:
It makes things more complicate. I've developed the following, which is a lot more complex code then a simple `Map<String, String>`:

```java
var credentials = context.getConfig(BASIC_AUTH).partition()
                .flatMap(it -> it.getRelativeEntries().entrySet().stream())
                .map(entry -> new Credentials(entry.getKey(), entry.getValue()))
                .collect(Collectors.toList());
...
static class Credentials {
    String username;
    String vaultKey;

    Credentials(String username, String vaultKey) {
        this.username = username;
        this.vaultKey = vaultKey;
    }
}
```

Will ignore this for the moment, because this is one of the PRs Catena-X needs until tomorrow :-/